src/scalar.rs

```diff
@@ -581,12 +581,12 @@ impl Scalar {
     /// # fn main() {
     /// let mut h = Sha512::default();
     ///
-    /// h.input(b"To really appreciate architecture, you may even need to commit a murder.");
-    /// h.input(b"While the programs used for The Manhattan Transcripts are of the most extreme");
-    /// h.input(b"nature, they also parallel the most common formula plot: the archetype of");
-    /// h.input(b"murder. Other phantasms were occasionally used to underline the fact that");
-    /// h.input(b"perhaps all architecture, rather than being about functional standards, is");
-    /// h.input(b"about love and death.");
+    /// h.input("To really appreciate architecture, you may even need to commit a murder.");
+    /// h.input("While the programs used for The Manhattan Transcripts are of the most extreme");
+    /// h.input("nature, they also parallel the most common formula plot: the archetype of");
+    /// h.input("murder. Other phantasms were occasionally used to underline the fact that");
+    /// h.input("perhaps all architecture, rather than being about functional standards, is");
+    /// h.input("about love and death.");
```

Comment:
This was fun. The `input()` method now takes `AsRef<[u8]>` as a parameter, however bytestring literals are `Sized` and `AsRef` is only impl'd up to `[T; 32]`. Switching from bytestrings to regular strings gets us the `AsRef<[u8]>` impl we need. Hax, but it's just an example...

Reply:
You also can demonstrate the new `chain` method by rewriting it as:
```Rust
let h = Sha256::new()
    .chain("To really appreciate architecture, you may even need to commit a murder.");
    .chain("While the programs used for The Manhattan Transcripts are of the most extreme");
    .chain("nature, they also parallel the most common formula plot: the archetype of");
    .chain("murder. Other phantasms were occasionally used to underline the fact that");
    .chain("perhaps all architecture, rather than being about functional standards, is");
    .chain("about love and death.");
```